### PR TITLE
Laboratory work #1, Golovleva Yekaterina - 20FPL3

### PR DESCRIPTION
**Pull request by esgolovleva**

### CODE REVIEW

**Comment by WhiteJaeger** [comment]
It's time to push some changes. October 1st is coming

**Comment by WhiteJaeger** [comment]
It's time to push some changes. October 1st is coming.

**Comment by esgolovleva** [comment]
Some style mistakes cannot be fixed!

File: lab_1/main.py [hunk 1]
```diff
@@ -51,7 +87,17 @@ def create_language_profile(language: str, text: str, stop_words: list) -> dict
     :param stop_words: a list of stop words
     :return: a dictionary with three keys – name, freq, n_words
     """
-    pass
+    if not isinstance(language, str):
+        return None
+    if not isinstance(text, str):
+        return None
+    if not isinstance(stop_words, list):
+        return None
```
Comment (WhiteJaeger) [inline lab_1/main.py:95]: These checks could be combined.

File: lab_1/main.py [hunk 2]
```diff
@@ -62,10 +108,24 @@ def compare_profiles(unknown_profile: dict, profile_to_compare: dict, top_n: int
     :param top_n: a number of the most common words
     :return: the distance
     """
-    pass
-
-
-def detect_language(unknown_profile: dict, profile_1: dict, profile_2: dict, top_n: int) -> str or None:
```
Comment (WhiteJaeger) [inline lab_1/main.py:68]: Why change the interface?

File: lab_1/main.py [hunk 3]
```diff
@@ -62,10 +108,24 @@ def compare_profiles(unknown_profile: dict, profile_to_compare: dict, top_n: int
...
+    if not isinstance(unknown_profile, dict):
+        return None
+    if not isinstance(profile_to_compare, dict):
+        return None
+    if not isinstance(top_n, int):
+        return None
```
Comment (WhiteJaeger) [inline lab_1/main.py:116]: Could be merged.

File: lab_1/main.py [hunk 4]
```diff
@@ -62,10 +108,24 @@ def compare_profiles(unknown_profile: dict, profile_to_compare: dict, top_n: int
...
+    common_top_n = 0
+    for i in unknown_profile_top_n:
+        if i in profile_to_compare_top_n:
+            common_top_n = common_top_n + 1
+    proportion = common_top_n/len(profile_to_compare_top_n)
+    proportion = round(proportion, 2)
+    return proportion
+
+
+def detect_language(unknown_profile: dict, profile_1: dict, profile_2: dict, top_n: int):
```
Comment (WhiteJaeger) [inline lab_1/main.py:128]: Why change the interface?

File: lab_1/main.py [hunk 5]
```diff
@@ -74,10 +134,25 @@ def detect_language(unknown_profile: dict, profile_1: dict, profile_2: dict, top
     :param top_n: a number of the most common words
     :return: a language
     """
-    pass
-
-
-def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, top_n: int) -> list or None:
```
Comment (WhiteJaeger) [inline lab_1/main.py:80]: Why change the interface?

File: lab_1/main.py [hunk 6]
```diff
@@ -74,10 +134,25 @@ def detect_language(unknown_profile: dict, profile_1: dict, profile_2: dict, top
...
+        return name_1
+    if profile_2_tokens > profile_1_tokens:
+        return name_2
+    if profile_1_tokens == profile_2_tokens:
+        names = [name_1, name_2]
+        names = sorted(names)
+        return names
+
+
+def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, top_n: int):
```
Comment (WhiteJaeger) [inline lab_1/main.py:155]: Why change the interface?

File: lab_1/main.py [hunk 7]
```diff
@@ -86,10 +161,38 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
...
+              'common': common_top_n,
+              'score': score,
+              'max_length_word': max_length_word,
+              'min_length_word': min_length_word,
+              'average_token_length': average_token_length,
+              'sorted_common': sorted_common}
+    return report
+
+
+def detect_language_advanced(unknown_profile: dict, profiles: list, languages: list, top_n: int):
```
Comment (WhiteJaeger) [inline lab_1/main.py:195]: Why change the interface?

File: lab_1/main.py [hunk 8]
```diff
@@ -98,7 +201,33 @@ def detect_language_advanced(unknown_profile: dict, profiles: list, languages: l
...
+        return None
+    if not isinstance(languages, list) or not isinstance(top_n, int):
+        return None
+    dictionary_score = {}  # {'name': score}
+    for profile in profiles:
+        if not languages or profile['name'] in languages:
+            profile_compare = compare_profiles_advanced(unknown_profile, profile, top_n)
+            dictionary_score[profile['name']] = profile_compare['score']
+    scores = list(dictionary_score.values())
+    if len(scores) == 0:
```
Comment (WhiteJaeger) [inline lab_1/main.py:214]: Could be simplified.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]
Good overall yet you should clean up the `main.py` file. 

`i`, `x` and such are not very descriptive names for a variable, are they?

**Comment by esgolovleva** [comment]
1. Variables have been renamed with more descriptive names.
2. The function descriptions are back in place (interface).
3. The checks for the correctness of the entered data have been combined.
4. The part of the code that needed to be simplified has been simplified.

**Review by WhiteJaeger — APPROVED** [review]

**Pull request closed without merging**